### PR TITLE
Add Docs Site

### LINKED ISSUE #376: Build a real docs website
See open zeppelin's for how this can be done well and generated from solidity: https://v2-3-0-rc-1--openzeppelin-docs-next.netlify.com/get-started. They use a package called `solidity-docgen`.

### PR DESCRIPTION
**Pull request by mrice32**
This PR adds a full autogenerated documentation static site modeled after docs.openzeppelin.com. It depends on two forks of open zeppelin work:

- https://github.com/UMAprotocol/docs_ui, which is a new repo because open zeppelin did not isolate their UI components from the rest of their documentation website. Loosely based on https://github.com/OpenZeppelin/docs.openzeppelin.com. However, I made quite a few changes to make the UI more UMA-y :).
- https://github.com/UMAprotocol/solidity-docgen/tree/fork, which is a fork of open zeppelin's repo by the same name. We've been forking this repository for a while, but I made more significant changes to make this work better for our needs. Two changes of note: I fixed a bug where array types don't display correctly (opened an issue about this on their repo) and I made overriding functions/events/modifiers append the base class documentation (this is important for cases where we only document the interface and not the derived class).

Multi contract files, like TokenizedDerivative.sol, still don't work. To get this to work, we'll need to split that file apart. Since this is a pretty easy change on our end, I figure it's easier to do this than making modifications in the library to handle multi-contract files.

In summary, this PR:
- Adds a `documentation/` folder where most documentation-related files are housed. Miscellaneous files were added there to interact with antora, handlebars, and the two external repos.
- Adds a script to build the documentation site.
- Adds a GAE config and a script to deploy the documentation site to GAE.
- Adds the docs site deployment to the `deploy_to_staging` stage in ci.

To preview the site, go to the artifacts tab under a successfully run docs ci job. 

Fixes #376.

### CODE REVIEW

File: playbook.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    google_analytics: 'UA-130599982-5'
+  start_page: uma::index.adoc
+content:
+  sources:
+    - url: .
+      branches: HEAD
+# ui:
+#   bundle:
+#     url: https://gitlab.com/antora/antora-ui-default/-/jobs/artifacts/master/raw/build/ui-bundle.zip?job=bundle-stable
+#     snapshot: true
```
[the remark this answers is not in the document]
Comment (mrice32) [reply playbook.yml]: Done.
